src/backend/latex/document/thesis.rs

```diff
@@ -127,17 +127,28 @@ impl<'a> Backend<'a> for Thesis {
         Ok(())
     }
 
-    fn gen_epilogue(&mut self, _cfg: &Config, out: &mut impl Write, _stderr: Arc<Mutex<StandardStream>>) -> FatalResult<()> {
+    fn gen_epilogue(&mut self, _cfg: &Config, out: &mut impl Write, _diagnostics: &Diagnostics<'a>) -> FatalResult<()> {
         writeln!(out, "\\end{{document}}")?;
         Ok(())
     }
 }
 
-fn gen(path: PathBuf, cfg: &Config, out: &mut impl Write, stderr: Arc<Mutex<StandardStream>>) -> FatalResult<()> {
+fn gen_abstract(path: PathBuf, abstract_name: &str, cfg: &Config, out: &mut impl Write, diagnostics: &Diagnostics<'_>) -> FatalResult<()> {
     let arena = Arena::new();
+    let stderr = Arc::clone(diagnostics.stderr());
     let mut gen = Generator::new(cfg, Thesis, out, &arena, stderr);
     let markdown = fs::read_to_string(&path)?;
-    let context = Context::LocalAbsolute(path.clone());
+    let context = match Context::from_path(path.clone()) {
+        Ok(context) => context,
+        Err(e) => {
+            diagnostics
+                .error(format!("invalid path to `{}`", abstract_name))
```

Comment:
This error could still be improved. There is very little context available for the user, a it is missing span information and does not mention that the cause is the resolution of the abstract file. I'm not sure if 'invalid path' is the best wording but can't find any better, it suffices as being reasonably clear with the note.

Reply:
I hope that I clarified it a little better.